app/helpers/compute_resources_vms_helper.rb

```diff
@@ -142,8 +142,12 @@ def show_vm_name?
     controller_name != 'hosts' && controller_name != 'compute_attributes'
   end
 
-  def new_host?(host)
-    host.try(:new_record?)
+  def new_host?(host, options = {})
+    if host.nil? && options[:nil_is_new]
```

Comment:
Which #new and #edit contexts do you mean?  I don't think it's used in the VM attributes partials.